tags/tags-impl/impl/src/java/org/sakaiproject/tags/impl/job/TagsExportedXMLSyncJob.java

```diff
@@ -0,0 +1,222 @@
+ *
+ *
+ */
+public class TagsExportedXMLSyncJob extends TagSynchronizer implements Job {
+	private static final Log log = LogFactory.getLog(TagsExportedXMLSyncJob.class);
+
+	private String tagsPathToXml;
+	private String tagCollectionsPathToXml;
+
+	private ServerConfigurationService serverConfigurationService;
+	private EmailService emailService;
+
+	private String collectionToUpdate=null;
+	private Boolean collectionToUpdateMoreThanOne=false;
+
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public InputStream getTagsXmlInputStream() {
+		tagsPathToXml = serverConfigurationService.getSakaiHomePath() + serverConfigurationService.getString("tags.fullxmltagsfile","tags/fullxmltags.xml");
+		log.debug(tagsPathToXml);
+		File xmlFile = new File(tagsPathToXml);
+		InputStream targetStream = null;
+				try {
+					targetStream=new FileInputStream(xmlFile);
+				} catch (Exception e){
+					log.warn("The Full Tags XML file can't be found in the specified route: " + tagsPathToXml);
+					e.printStackTrace();
+				}
+
+		return targetStream;
+
+
+	}
+
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public void execute(JobExecutionContext arg0) throws JobExecutionException {
+
+		syncAllTags();
+	}
+
+
+	public synchronized void syncAllTags() {
+		long start = System.currentTimeMillis();
+		if(log.isInfoEnabled()) log.info("Starting Full XML Tag Collection synchronization");
+		// Load the xml document from the xml stream
+
+		XMLReader tagsReader = new XMLReader();
+		// We can add several handlers which are triggered for a given node
+		// name. The complete sub-dom of this node is then parsed and made
+		// available as a StructuredNode
+
+
+		tagsReader.addHandler("tagservice-admin", new NodeHandler() {
+
+
+			@Override
+			public void process(StructuredNode node) {
+				try {
+
+					String tagLabel = node.queryString("tagLabel");
+					String tagId = node.queryString("tagId");
+					String externalId = node.queryString("externalId");
+					String description = node.queryString("description");
+					long externalCreationDate = node.queryValue("externalCreationDate").asLong(0L);
+					long lastUpdateDateInExternalSystem = node.queryValue("lastUpdateDateInExternalSystem").asLong(0L);
+					String externalHierarchyCode = node.queryString("externalHierarchyCode");
+					String externalType = node.queryString("externalType");
+					String alternativeLabels = node.queryString("alternativeLabels");
+					String tagCollectionId =  node.queryString("tagCollectionId");
+					//We will check here if we have one and only one collectionid in all the tags to update.
+					if (collectionToUpdate!=null && !collectionToUpdateMoreThanOne){
+						if (!(collectionToUpdate.equals(tagCollectionId))){
+							collectionToUpdateMoreThanOne=true;
+						}
+					}else{
+						collectionToUpdate=tagCollectionId;
+					}
+					String data =  node.queryString("data");
+					String parentId = node.queryString("parentId");
+
+					if (tagId !=null && tagId.length()==36){
```

Comment:
36 is the length of a valid ID. Just a small extra check to be sure the file has good ids... If we have a valid ID then we use it if not we create a new tag. I will change that to a static constant.